protocols/noise/src/io.rs

```diff
@@ -249,13 +268,21 @@ impl<T: io::Write> io::Write for NoiseOutput<T> {
                     }
                     return Ok(n)
                 }
-                WriteState::WriteLen { len } => {
-                    trace!("write: writing len ({})", len);
-                    if !write_frame_len(&mut self.io, len as u16)? {
-                        trace!("write: eof");
-                        self.write_state = WriteState::Eof;
-                        return Err(io::ErrorKind::WriteZero.into())
-                    }
+                WriteState::WriteLen { len, mut buf, mut off } => {
+                    trace!("write: writing len ({}, {:?}, {}/2)", len, buf, off);
+                    match write_frame_len(&mut self.io, &mut buf, &mut off) {
+                        Err(e) => {
+                            if e.kind() == io::ErrorKind::WouldBlock {
+                                self.write_state = WriteState::WriteLen{ len, buf, off };
+                            }
+                            return Err(e)
+                        }
+                        Ok(false) => {
+                            trace!("write: eof");
+                            self.write_state = WriteState::Eof;
+                            return Err(io::ErrorKind::WriteZero.into())
+                        }
+                        Ok(true) => ()                    }
```

Comment:
The curly brace should go on the next line I think.

Reply:
Oops!